safekeeper/src/safekeeper.rs

```diff
@@ -634,7 +634,8 @@ where
         }
 
         // system_id will be updated on mismatch
-        if self.state.server.system_id != msg.system_id {
+        // sync-safekeepers doesn't know sysid and sends 0, ignore it
+        if self.state.server.system_id != msg.system_id && msg.system_id != 0 {
```

Comment:
(Unrelated NIT): Below on line 649 we update `pg_version` in control file, that's very much likely a dead code, we can delete that in another PR